### PR TITLE
Document using a certificate with a full chain

### PR DESCRIPTION
**Pull request by anoadragon453**
Closes https://github.com/matrix-org/synapse/issues/4764

Tell people to use a full chain certificate instead of just the top-level one (which browsers will accept, but clients not packing their own certificates will not).

While we mentioned that we don't explicitly include instructions for certbot/Let's Encrypt, I believe as it's the dominant way people are getting their certificates, including a line to help those people is essential, and means less tech support to do down the line.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -374,9 +374,12 @@ To configure Synapse to expose an HTTPS port, you will need to edit
...
   `tls_private_key_path` lines under the `TLS` section. You can either
   point these settings at an existing certificate and key, or you can
-  enable Synapse's built-in ACME (Let's Encrypt) support.  Instructions
-  for having Synapse automatically provision and renew federation 
-  certificates through ACME can be found at [ACME.md](docs/ACME.md).
+  enable Synapse's built-in ACME (Let's Encrypt) support. Instructions
+  for having Synapse automatically provision and renew federation
+  certificates through ACME can be found at [ACME.md](docs/ACME.md). If you
+  are using your own certificate and key, be sure to use a certificate that
+  includes a full chain (for instance, if using Let's Encrypt, use
```
Comment (babolivier) [inline INSTALL.md]:
```suggestion
  includes a full chain (for instance, if using certbot, use
```

Comment (anoadragon453) [reply INSTALL.md]: Sure

File: INSTALL.md [hunk 2]
```diff
@@ -375,9 +375,12 @@ To configure Synapse to expose an HTTPS port, you will need to edit
 * You will also need to uncomment the `tls_certificate_path` and
   `tls_private_key_path` lines under the `TLS` section. You can either
   point these settings at an existing certificate and key, or you can
-  enable Synapse's built-in ACME (Let's Encrypt) support.  Instructions
+  enable Synapse's built-in ACME (Let's Encrypt) support. Instructions
   for having Synapse automatically provision and renew federation
-  certificates through ACME can be found at [ACME.md](docs/ACME.md).
+  certificates through ACME can be found at [ACME.md](docs/ACME.md). If you
+  are using your own certificate and key, be sure to use a certificate that
```
Comment (richvdh) [inline INSTALL.md]: wait, this is misworded. A certificate can't include a chain: you need the full chain *of* certificates.

"use a `.pem` file which includes the full certificate chain including any intermediate certificates".

Likewise for the config text,

**Comment by anoadragon453** [comment]
Apologies for the force-push, was clearing out some things. CI should hopefully be happy now.

**Pull request merged**